### PR TITLE
make package embeddable

### PR DESCRIPTION
**Pull request by stevegt**
I did some refactoring to make the package embeddable -- essentially move main to cmd/porcelain and export some accessor functions.  Depends on my earlier fix_tests PR.  You'll want to revert my temporary package paths in imports and go.mod before you merge this PR.

### CODE REVIEW

File: cmd/porcelain/main.go [hunk 1]
```diff
@@ -7,6 +7,9 @@ import (
 	"io/ioutil"
 	"log"
 	"os"
+
+	// "github.com/robertgzr/porcelain"
+	"github.com/stevegt/porcelain"
```
Comment (stevegt) [inline cmd/porcelain/main.go:12]: revert before merging

File: go.mod [hunk 2]
```diff
@@ -1,4 +1,5 @@
-module github.com/robertgzr/porcelain
+// module github.com/robertgzr/porcelain
+module github.com/stevegt/porcelain
```
Comment (stevegt) [inline go.mod:2]: revert before merging

File: cmd/porcelain/main.go (hunk 1 above)
Comment (robertgzr) [reply cmd/porcelain/main.go:12]: can you revert those changes from the pr branch?

**Review by robertgzr — CHANGES_REQUESTED** [review]
If you could change the imports and go.mod that would be great.

and possibly fix the makefile, but I can do that in a later commit as well:
```diff
diff --git a/Makefile b/Makefile
index 00c4f9e..6b60dab 100644
--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,7 @@ GO_LDFLAGS += -X main.date=$(shell date -u -I) -X main.commit=$(REV) -X main.ver
 all: porcelain porcelain.1
 
 porcelain:
-	$(GO) build $(GOFLAGS) -ldflags "$(GO_LDFLAGS)"
+	$(GO) build $(GOFLAGS) -ldflags "$(GO_LDFLAGS)" ./cmd/$@
 
 porcelain.1: porcelain.1.scd
 	$(SCDOC) < $< >$@
```

File: go.mod (hunk 2 above)
Comment (stevegt) [reply go.mod]: done

File: cmd/porcelain/main.go (hunk 1 above)
Comment (stevegt) [reply cmd/porcelain/main.go]: done

**Comment by robertgzr** [comment]
Thanks o/

**Pull request closed without merging**